### PR TITLE
fix: always force x64 arch for `flutter build windows`

### PR DESCRIPTION
**Pull request by plarson**
Attempting to build a Flutter project on Windows with an `ARM64` processor will lead to linker errors since Flutter only targets `x64` DLLs. This simply pins the CMake generator to `x64` so you can still build when running with an M1 Mac using Parallels or any other ARM based machine.

Projects with pre-existing CMake caches in `/build` need to be regenerated.

[Add Support for running Flutter on Windows ARM64](https://github.com/flutter/flutter/issues/53120)

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I signed the [CLA].
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I added new tests to check the change I am making, or this PR is [test-exempt].
- [X] All existing and new tests are passing.

### CODE REVIEW

**Comment by christopherfujino** [comment]
Why is this test exempt?

**Comment by christopherfujino** [comment]
> It doesn't build on ARM64 so a test would fail anyway and x64 tests should remain the same. Happy to remove.

Well, this discussion is moot, since you'll need to fix broken tests: https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8780839964683901601/+/u/run_test.dart_for_tool_tests_shard_and_subshard_commands/test_stdout

**Comment by plarson** [comment]
Got it

**Comment by christopherfujino** [comment]
@loic-sharma should this be a CLI flag?

File: packages/flutter_tools/test/commands.shard/hermetic/build_windows_test.dart [hunk 1]
```diff
@@ -85,6 +85,8 @@ void main() {
         r'build\windows',
         '-G',
         generator,
+		'-A',
```
Comment (christopherfujino) [inline packages/flutter_tools/test/commands.shard/hermetic/build_windows_test.dart:88]: please use spaces not tabs, and align with the previous lines

Comment (plarson) [reply packages/flutter_tools/test/commands.shard/hermetic/build_windows_test.dart]: Changed my editor

**Comment by loic-sharma** [comment]
Hello, I'm out this week. I should be able to take a closer look at this next week. However, I believe changing CMake's generate arguments will require users to `flutter clean` once before they can build again.

There's a [design doc for Arm64 support on Windows](https://docs.google.com/document/d/1yuexY-EtzeVhU3V6RGBII_kH-znLSkQCspzW70jayZk/edit?usp=sharing&resourcekey=0-8pQRSXSHQlwOo7QtCVLvBA). It avoids this particular problem as it moves the build directory from `build/windows/` to `build/windows/arm64`.

**Comment by loic-sharma** [comment]
Hello, I'll close this off for now as we'll need to update the build layout before we can specify CMake's architecture. For more information, please refer to the [design doc for Arm64 support on Windows](https://docs.google.com/document/d/1yuexY-EtzeVhU3V6RGBII_kH-znLSkQCspzW70jayZk/edit?usp=sharing&resourcekey=0-8pQRSXSHQlwOo7QtCVLvBA). Thank you for your contribution!

**Pull request closed without merging**